tensorboard/webapp/webapp_data_source/tb_feature_flag_data_source_test.ts

```diff
@@ -247,5 +247,172 @@ describe('tb_feature_flag_data_source', () => {
         );
       });
     });
+
+    describe('persistFeatureFlags', () => {
+      it('calls getPersistentFeatureFlags and stores new flag', () => {
+        const getPersistentFeatureFlagsSpy = spyOn(
+          dataSource,
+          'getPersistentFeatureFlags'
+        ).and.callThrough();
+
+        spyOn(localStorage, 'getItem').and.returnValue(null);
+
+        const setItemSpy = spyOn(localStorage, 'setItem').and.stub();
+
+        dataSource.persistFeatureFlags({enabledScalarDataTable: true});
+
+        expect(getPersistentFeatureFlagsSpy).toHaveBeenCalled();
+        expect(setItemSpy).toHaveBeenCalledOnceWith(
+          'tb_feature_flag_storage_key',
+          '{"enabledScalarDataTable":true}'
+        );
+      });
+
+      it('adds new flag when some flags already exist', () => {
+        const getPersistentFeatureFlagsSpy = spyOn(
+          dataSource,
+          'getPersistentFeatureFlags'
+        ).and.callThrough();
+
+        spyOn(localStorage, 'getItem').and.returnValue(
+          '{"enabledScalarDataTable":true}'
+        );
+
+        const setItemSpy = spyOn(localStorage, 'setItem').and.stub();
+
+        dataSource.persistFeatureFlags({enableTimeSeriesPromotion: true});
```

Comment:
Maybe in this test case you could set more than 1 flag at a time just to make sure that works. (Or add a new test for that).

That is, this line could be:
```
        dataSource.persistFeatureFlags({enableTimeSeriesPromotion: true, enableLinkedTime: true, enableDataTable: true});
```

Reply:
I added a new test for multiple flags.